File: backend/app/schemas/escalation_template.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.base import Str50, Str2000, TextStr


class EscalationTemplateCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    category: Str50
    severity: Str50
    description_template: Str2000 | None = None
    suggested_actions: list[str] | None = None
    notification_template: TextStr | None = None
    is_active: bool = True


class EscalationTemplateUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    category: Str50 | None = None
    severity: Str50 | None = None
    description_template: Str2000 | None = None
    suggested_actions: list[str] | None = None
    notification_template: TextStr | None = None
    is_active: bool | None = None


class EscalationTemplateResponse(BaseModel):
    id: UUID
    name: str = Field(..., max_length=255)
    category: Str50
    severity: Str50
    description_template: Str2000 | None = None
    suggested_actions: list[str] | None = None
    notification_template: TextStr | None = None
    is_system: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class EscalationTemplateListResponse(BaseModel):
    templates: list[EscalationTemplateResponse]
    total: int

File: frontend/src/types/escalation-template.ts
/**
 * Escalation template types — re-exported from generated OpenAPI types where possible.
 *
 * @see backend/app/schemas/escalation_template.py
 */
import type { components } from "./generated";

// ---------------------------------------------------------------------------
// API types — re-exported from generated OpenAPI schema
// ---------------------------------------------------------------------------

export type EscalationTemplate = components["schemas"]["EscalationTemplateResponse"];
export type EscalationTemplateListResponse = components["schemas"]["EscalationTemplateListResponse"];

// ---------------------------------------------------------------------------
// Frontend-only types — enums, request shapes, constants
// ---------------------------------------------------------------------------

export type EscalationTemplateCategory =
  | "partner_sla_breach"
  | "client_dissatisfaction"
  | "resource_unavailable"
  | "budget_overrun"
  | "timeline_delay"
  | "quality_issue";

export type EscalationTemplateSeverity =
  | "task"
  | "milestone"
  | "program"
  | "client_impact";

export interface EscalationTemplateCreate {
  name: string;
  category: EscalationTemplateCategory;
  severity: EscalationTemplateSeverity;
  description_template?: string;
  suggested_actions?: string[];
  notification_template?: string;
  is_active?: boolean;
}

export interface EscalationTemplateUpdate {
  name?: string;
  category?: EscalationTemplateCategory;
  severity?: EscalationTemplateSeverity;
  description_template?: string;
  suggested_actions?: string[];
  notification_template?: string;
  is_active?: boolean;
}

export interface EscalationTemplateListParams {
  skip?: number;
  limit?: number;
  category?: EscalationTemplateCategory;
  severity?: EscalationTemplateSeverity;
  is_active?: boolean;
  is_system?: boolean;
}

export const CATEGORY_LABELS: Record<EscalationTemplateCategory, string> = {
  partner_sla_breach: "Partner SLA Breach",
  client_dissatisfaction: "Client Dissatisfaction",
  resource_unavailable: "Resource Unavailable",
  budget_overrun: "Budget Overrun",
  timeline_delay: "Timeline Delay",
  quality_issue: "Quality Issue",
};

export const SEVERITY_LABELS: Record<EscalationTemplateSeverity, string> = {
  task: "Task",
  milestone: "Milestone",
  program: "Program",
  client_impact: "Client Impact",
};
